### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by cjam**
- This one has quite a lots of changes.  Apologies.  

Some notable changes have occurred in the API.ts around the shape.  I have changed WorkSectionId type to `WorkSectionCode` as it is more fitting (since that code also now has an ID associated with it in the API).

Also, I think I'd like to establish a new pattern for our Forms that I think could improve maintability.  You can see it here:

https://github.com/bcgov/jag-shuber-frontend/compare/master...api-integration#diff-c24785101e8bb8848547edf051fad806R132

You can see it in action (why it's valuable) here:

OnSubmit -> https://github.com/bcgov/jag-shuber-frontend/compare/master...api-integration#diff-79b9b2bc13626d7abe349d835511841fR29

InitialValues -> https://github.com/bcgov/jag-shuber-frontend/compare/master...api-integration#diff-79b9b2bc13626d7abe349d835511841fR42

### CODE REVIEW

File: src/app/containers/AssignmentDropTarget.tsx [hunk 1]
```diff
@@ -1,43 +1,43 @@
-import * as React from 'react'
-import { DraggedAssignment, AssignmentDropResult } from './AssignmentDragSource'
-import { CSSProperties } from 'react';
-import dropTargetFactory from '../infrastructure/DragDrop/dropTargetFactory';
-import ItemTypes from '../infrastructure/DragDrop/ItemTypes';
+// import * as React from 'react'
```
Comment (hollyegrayson) [inline src/app/containers/AssignmentDropTarget.tsx]: Can we remove this file?

File: src/app/modules/sheriffs/actions.ts [hunk 2]
```diff
@@ -1,88 +1,96 @@
...
-}
+import * as sheriffRequests from './requests';
+
+
+export const getSheriffList = sheriffRequests.sheriffMapRequest.actionCreator;
+export const updateSheriff = sheriffRequests.updateSheriffRequest.actionCreator;
+export const createSheriff = sheriffRequests.createSheriffRequest.actionCreator;
+
+
+// import { Action } from 'redux';
```
Comment (hollyegrayson) [inline src/app/modules/sheriffs/actions.ts]: Clean this up?

File: src/app/modules/sheriffs/reducer.ts [hunk 3]
```diff
@@ -1,58 +1,84 @@
...
-  SHERIFF_UPDATE_SUCCESS: (state,payload)=>{
-    const newMap = Object.assign({},state.map);
-    newMap[payload.badgeNumber]=payload;
-    return {updating:false,map:newMap};
-  },
-  SHERIFF_UPDATE_FAIL: (state,payload)=>({updating:false,error:payload})
-});
 
-export default reducer;
+// import {
```
Comment (hollyegrayson) [inline src/app/modules/sheriffs/reducer.ts]: Clean this up?

File: src/app/api/Client.ts [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+// All of our resources now have an idPath
+// so we'll specify a HalResource that contains that
+class APIResource extends HalResource {
+    @HalProperty()
+    idPath: string;
+}
+
+export function extractWorksectionCode(workSectionCodePath: string): WorkSectionCode {
+    const code = `${workSectionCodePath}`.split('/').slice(-1)[0] as any;
+    return code !== '' ? code : 'COURTS';
```
[the remark this answers is not in the document]
Comment (cjam) [reply src/app/api/Client.ts]: 👍

File: src/app/components/AssignmentForm.tsx [hunk 5]
```diff
@@ -102,11 +104,59 @@ export interface AssignmentFormProps {
...
+    id?: string;
+    daysBitmap: DaysOfWeek;
+    sheriffsRequired: number;
+    timeRange: {
+        startTime: TimeType;
+        endTime: TimeType;
+    };
+}
+
+const TimeFormat = 'HH:mm:ss';
```
[the remark this answers is not in the document]
Comment (cjam) [reply src/app/components/AssignmentForm.tsx]: yea, I think this should probably actually live somewhere else in our utilities / infrastructure.  But I had it capitalized as it's a constant String. But I think our convention is all caps, so i'll change to that. 👍

File: src/app/containers/AssignmentDropTarget.tsx (hunk 1 above)
Comment (cjam) [reply src/app/containers/AssignmentDropTarget.tsx]: 👍

**Pull request merged**